src/layers/attention/self_attention.rs

```diff
@@ -280,15 +307,35 @@ pub enum SelfAttentionError {
     #[snafu(display("Cannot intersect attention and causal mask"))]
     IntersectMasks { source: AttentionMaskError },
 
+    #[snafu(display(
+        "The hidden size ({hidden_width}) must be divisble by the number of query heads ({n_query_heads})"
+    ))]
+    InvalidNHeads {
```

Comment:
```suggestion
    InvalidNQueryHeads {
```
There are arguments for both, I think. Your call.

Reply:
Yeah, that better. Fixed.